src/blocks/rofication.rs

```diff
@@ -0,0 +1,195 @@
+use std::str::FromStr;
+use std::time::Duration;
+
+use crossbeam_channel::Sender;
+use serde_derive::Deserialize;
+
+use std::io::Read;
+use std::io::Write;
+use std::os::unix::net::UnixStream;
+use std::path::Path;
+
+use crate::blocks::{Block, ConfigBlock, Update};
+use crate::config::SharedConfig;
+use crate::de::deserialize_duration;
+use crate::errors::*;
+use crate::protocol::i3bar_event::I3BarEvent;
+use crate::protocol::i3bar_event::MouseButton;
+use crate::scheduler::Task;
+use crate::subprocess::spawn_child_async;
+use crate::widgets::text::TextWidget;
+use crate::widgets::I3BarWidget;
+use crate::widgets::State;
+
+#[derive(Debug)]
+struct RotificationStatus {
+    num: u64,
+    crit: u64,
+}
+
+pub struct Rofication {
+    id: usize,
+    text: TextWidget,
+    update_interval: Duration,
+
+    //useful, but optional
+    #[allow(dead_code)]
+    shared_config: SharedConfig,
+    #[allow(dead_code)]
+    tx_update_request: Sender<Task>,
+    // UNIX socket to read from
+    pub socket_path: String,
+}
+
+#[derive(Deserialize, Debug, Clone)]
+#[serde(deny_unknown_fields, default)]
+pub struct RoficationConfig {
+    /// Update interval in seconds
+    #[serde(deserialize_with = "deserialize_duration")]
+    pub interval: Duration,
+    // UNIX socket to read from
+    pub socket_path: String,
+}
+
+impl Default for RoficationConfig {
+    fn default() -> Self {
+        Self {
+            interval: Duration::from_secs(1),
+            socket_path: String::from_str("/tmp/rofi_notification_daemon").unwrap(),
+        }
+    }
+}
+
+impl ConfigBlock for Rofication {
+    type Config = RoficationConfig;
+
+    fn new(
+        id: usize,
+        block_config: Self::Config,
+        shared_config: SharedConfig,
+        tx_update_request: Sender<Task>,
+    ) -> Result<Self> {
+        let text = TextWidget::new(id, 0, shared_config.clone())
+            .with_text("0")
```

Comment:
Should this just be a blank string `""` or `"?"` instead since we don't have the actual count yet?

Reply:
Changed to ?.